crates/aggchain-proof-core/src/full_execution_proof.rs

```diff
@@ -79,25 +104,20 @@ impl FepInputs {
 }
 
 #[repr(u8)]
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, PartialEq)]
 enum OptimisticMode {
     Sp1 = 0,
     Ecdsa = 1,
 }
 
 impl FepInputs {
+    pub fn encoded_aggchain_params(&self) -> Vec<u8> {
+        AggchainParamsValues::abi_encode_packed(&AggchainParamsValues::from(self))
+    }
+
     /// Compute the chain-specific commitment forwarded to the PP.
     pub fn aggchain_params(&self) -> Digest {
-        let optimistic_mode: u8 = self.optimistic_mode() as u8;
-
-        keccak256_combine([
-            self.compute_l2_pre_root().as_slice(),
-            self.compute_claim_root().as_slice(),
-            &self.claim_block_num.to_be_bytes(),
-            self.rollup_config_hash.as_slice(),
-            &optimistic_mode.to_be_bytes(),
-            self.trusted_sequencer.as_slice(),
-        ])
+        keccak256(self.encoded_aggchain_params().as_slice())
```

Comment:
Not strictly related to the PR but I think `aggchain_params_commitment` would be a more accurate name.

Reply:
ah `aggchain_params` as-is really refers to the hash itself same as in the specs and in the contracts